sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/ResourceIdentifier.cs

```diff
@@ -113,7 +119,11 @@ internal static ResourceIdentifier CreateBaseSubscriptionIdentifier(string subsc
                         return CreateBaseResourceGroupIdentifier(subscription, parts[1], parts.Trim(2));
                     case ProvidersKey:
                         {
-                            if (parts.Count > 3)
+                            if (parts.Count == 2 || parts[2] == ProvidersKey)
+                            {
+                                return CreateSubscriptionProviderIdentifier(new SubscriptionProviderIdentifier(new SubscriptionResourceIdentifier(), parts[1]), parts.Trim(2));
```

Comment:
can't use an empty constructor here, since that is how the Subscription property will be initialized